### ISSUE TITLE
Multiple query fail in specifics DNS servers

### ISSUE THREAD

**Issue by yhojann-cl** [martinheidegger/dns-query#2]
By example:

```javascript
query(
    {
        "questions": [
            {
                "type": "A",
                "name": "google.com"
            },
            {
                "type": "AAAA",
                "name": "google.com"
            }
        ]
    },
    {
        "endpoints": [
            {
                "protocol": "udp4:",
                "host": "127.0.0.53"
            }
        ],
        "retry": 3,
        "timeout": 4000
    }
).then(data => console.log(data.answers))
```

> UnhandledPromiseRejectionWarning: TimeoutError: Timeout (t=4000)

Fail using local DNS server (using ubuntu 20.04 LTS) but using `8.8.8.8` works fine, but using only one question works fine from local.

**Comment by martinheidegger** [comment]
This may be a problem with the DNS server that you are using _(many dns servers do not support multiple queries in a request)_ or in the underlying [dns-socket](https://github.com/mafintosh/dns-socket/) package as [the request is simply passed-on](https://github.com/martinheidegger/dns-query/blob/b1377fa87939c9feef27dc969aa09e034fb0f4ab/lib.node.js#L60). More debugging information might help here. Do you have the raw udp transport data? Can you test a request with multiple queries with a different client? What server do you use?

**Comment by WHK102** [comment]
Yes, here more data:

```bash
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.4 LTS
Release:	20.04
Codename:	focal
$ netstat -noa | grep 127.0.0.53:53
tcp        0      0 127.0.0.53:53           0.0.0.0:*               ESCUCHAR    apagado (0.00/0/0)
udp        0      0 127.0.0.53:53           0.0.0.0:*                           apagado (0.00/0/0)
$ sudo lsof -i :53
[sudo] contraseña para whk: 
COMMAND    PID            USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
systemd-r 1218 systemd-resolve   12u  IPv4  36906      0t0  UDP localhost:domain 
systemd-r 1218 systemd-resolve   13u  IPv4  36907      0t0  TCP localhost:domain (LISTEN)
dnsmasq   1629 libvirt-dnsmasq    5u  IPv4  37854      0t0  UDP machine:domain 
dnsmasq   1629 libvirt-dnsmasq    6u  IPv4  37855      0t0  TCP machine:domain (LISTEN)
$ ps -aux | grep systemd-resolve
systemd+    1218  0.0  0.0  24264  6484 ?        Ss   mar22   0:09 /lib/systemd/systemd-resolved
$ systemctl status systemd-resolved
● systemd-resolved.service - Network Name Resolution
     Loaded: loaded (/lib/systemd/system/systemd-resolved.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2022-03-22 13:55:27 -03; 1 weeks 0 days ago
       Docs: man:systemd-resolved.service(8)
             https://www.freedesktop.org/wiki/Software/systemd/resolved
             https://www.freedesktop.org/wiki/Software/systemd/writing-network-configuration-managers
             https://www.freedesktop.org/wiki/Software/systemd/writing-resolver-clients
   Main PID: 1218 (systemd-resolve)
     Status: "Processing requests..."
      Tasks: 1 (limit: 14088)
     Memory: 4.9M
     CGroup: /system.slice/systemd-resolved.service
             └─1218 /lib/systemd/systemd-resolved

mar 22 13:55:27 machine systemd[1]: Starting Network Name Resolution...
mar 22 13:55:27 machine systemd-resolved[1218]: Positive Trust Anchors:
...
$ cat /etc/resolv.conf 
nameserver 127.0.0.53
$ dig +short -t a google.com @127.0.0.53
142.251.0.139
142.251.0.113
142.251.0.101
142.251.0.100
142.251.0.102
142.251.0.138
```

By default linux systems with Network Manager like as Ubuntu use local DNS server for replication and local cache, this dns service gets records from ip gateway but web browsers, cli tools, desktop apps, etc use local dns service.

The package is https://manpages.ubuntu.com/manpages/xenial/en/man8/systemd-resolved.service.8.html

Try using a vm with Ubuntu and two questions in same query, the library does not work.

For me i solve this separate each query by record type resolution in a diferent thread:

```javascript
// Not all dns servers support multiple dns queries in a single packet:
// https://github.com/martinheidegger/dns-query/issues/2
recordsEntries
    .filter(([recordType, record]) => { return record.enabled })
    .map(([recordType, record]) => { return { type: recordType, name: hostname } })
    .forEach(question => {
        query({
            questions: [ question ]
        }, {
            endpoints: [{ protocol: 'udp4:', host: customNS }],
            retry: 3,
            timeout: 4000
        })
        .then(data => {
```

**Comment by martinheidegger** [comment]
Thank you for the debugging data. This seems like a harder bug and the workaround to do the requests one-by-one should work. If you _(or someone else reading this)_ has time to further investigate this, I would appreciate it.

**Comment by martinheidegger** [comment]
I read up on this and it turns out that it is not possible to properly implement multiple questions in one query. The currently released version `v0.9.1` (see: #3) only allows a single question per query. In a future version it may offer an API to do multiple questions in one request but that would be a bigger refactoring.
